code/modules/clothing/chameleon.dm

```diff
@@ -760,3 +763,168 @@
 /obj/item/clothing/neck/chameleon/broken/Initialize(mapload)
 	. = ..()
 	chameleon_action.emp_randomise(INFINITY)
+
+/obj/item/gun/energy/laser/chameleon
+	var/datum/action/item_action/chameleon/change/chameleon_action
+
+	ammo_type = list(/obj/item/ammo_casing/energy/chameleon)
+	pin = /obj/item/firing_pin
+	automatic_charge_overlays = FALSE
+	can_select = FALSE
+
+	/// The vars we want copied over to our projectile.
+	var/list/projectile_copy_vars
+	/// The vars copied over to our projectile on fire.
+	var/list/chameleon_projectile_vars
+
+	/// The vars we want copied over to our gun.
+	var/list/gun_copy_vars
+	/// The vars copied over to our projectile on selection.
+	var/list/chameleon_gun_vars
+
+	/// The vars we want copied over to our ammo.
+	var/list/ammo_copy_vars
+	/// The vars copied over to our ammo on selection.
+	var/list/chameleon_ammo_vars
```

Comment:
The use of vars here is not an acceptable pattern.
Is there any reason why you can't just set the variables directly?

Reply:
frankly there's too many variables to directly transcribe over to the chameleon projectile to be worth manually setting each one. more bloat than anything. frankly, we've used `vars[]` for less, and this is precisely the kind of implementation where its usage would be warranted: mass setting a large amount of variables. seeing as how everything getting their vars set is typepathed, directly inputted, and out of the player's control completely, it's completely reasonable to be using the property

Reply:
also: there is no practical, logical, or possible way a player could abuse this behavior to fuck with other more important vars. i believe this is a complete nonissue, proven otherwise. again, we've used it for less

Reply:
The threat vector is not players, it is admins.

> frankly there's too many variables to directly transcribe over to the chameleon projectile to be worth manually setting each one. more bloat than anything. 

Then you shouldn't do it.

> frankly, we've used vars[] for less, and this is precisely the kind of implementation where its usage would be warranted: mass setting a large amount of variables. 

Because my fellow maintainers have decided that security for some reason matters, we should not be using vars in cases like this, not without protection.

> seeing as how everything getting their vars set is typepathed, directly inputted, and out of the player's control completely, it's completely reasonable to be using the property

Again, the threat is not players, it is admins. 
The typepath of the variable being set does nothing. It could be anything. Types in dm are only enforced on variable access, and somewhat during compile. With what you have now I could set `loaded_projectile` to GLOB, and edit any global variable we have, including the one that prevents unsafe file accesses.

This pr should not be merged while this is an issue.

Reply:
1. Use `vv_edit_var`
2. Use `NAMEOF`. When the properties change, we want a compile error, and `NAMEOF` will do that.